Description: fix segfault with certain uses of XTest because of null pointer dereference.
Origin: upstream, http://cgit.freedesktop.org/xorg/xserver/commit/?id=ad48184752970541aa153bf2961a2d7da1a910b5
Ubuntu-Bug: https://bugs.launchpad.net/ubuntu/+source/xorg-server/+bug/607071

diff -Nur xorg-server-1.7.6/Xext/xtest.c xorg-server-1.7.6.new/Xext/xtest.c
--- xorg-server-1.7.6/Xext/xtest.c	2010-03-11 23:38:21.000000000 -0500
+++ xorg-server-1.7.6.new/Xext/xtest.c	2010-12-06 09:38:33.000000000 -0500
@@ -184,6 +184,7 @@
     int i;
     int base = 0;
     int flags = 0;
+    int need_ptr_update = 1;
 
     nev = (stuff->length << 2) - sizeof(xReq);
     if ((nev % sizeof(xEvent)) || !nev)
@@ -387,6 +388,8 @@
                 client->errorValue = ev->u.u.detail;
                 return BadValue;
             }
+
+            need_ptr_update = 0;
             break;
         case MotionNotify:
             if (!dev->valuator)
@@ -451,7 +454,8 @@
     for (i = 0; i < nevents; i++)
         mieqProcessDeviceEvent(dev, (InternalEvent*)(xtest_evlist+i)->event, NULL);
 
-    miPointerUpdateSprite(dev);
+    if (need_ptr_update)
+        miPointerUpdateSprite(dev);
     return client->noClientException;
 }
 
